Textfield/Textfield.jsx

```diff
@@ -3,49 +3,120 @@ import MaterialComponent from "../MaterialComponent";
 import { MDCTextfield } from "@material/textfield/";
 
 /**
+ * @prop persistent = false
+ * @prop validation-msg = false
+ */
+class Helptext extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "textfield-helptext";
+    this._mdcProps = ["persistent", "validation-msg"];
+  }
+  materialDom(props) {
+    return (
+      <p {...props} aria-hidden="true">
+        {props.children}
+      </p>
+    );
+  }
+}
+
+/**
  * @prop fullwidth = false
  * @prop multiline = false
+ * @prop dense = false
+ * @prop disabled = false
  * @prop type = 'text'
  */
-export default class Textfield extends MaterialComponent {
+class TextfieldInput extends MaterialComponent {
   constructor() {
     super();
     this.componentName = "textfield";
-    this.state = {
-      showFloatingLabel: false
-    };
-    this._mdcProps = ["fullwidth", "multiline"];
+    this._mdcProps = ["fullwidth", "multiline", "dense", "disabled"];
   }
   componentDidMount() {
-    this.setState({
-      showFloatingLabel: true
-    });
     this.MDComponent = new MDCTextfield(this.control);
   }
   componentWillUnmount() {
     this.MDComponent.destroy && this.MDComponent.destroy();
   }
   materialDom(allprops) {
     const { className, ...props } = allprops;
-    let labelClass = ["mdc-textfield__label"];
+
+    // Label
+    const labelClass = ["mdc-textfield__label"];
     if (props.value) {
       labelClass.push("mdc-textfield__label--float-above");
     }
+
     return (
       <div className={className} ref={control => (this.control = control)}>
         {props.multiline
-          ? <textarea className="mdc-textfield__input" {...props} />
+          ? <textarea
+              className="mdc-textfield__input"
+              aria-controls={props.id + "-helptext"}
+              {...props}
+            />
           : <input
               type={props.type || "text"}
               className="mdc-textfield__input"
+              aria-controls={props.id + "-helptext"}
               {...props}
             />}
         {props.label &&
-          this.state.showFloatingLabel &&
           <label className={labelClass.join(" ")}>
             {props.label}
           </label>}
       </div>
     );
   }
 }
+
+/**
+ * @prop fullwidth = false
+ * @prop multiline = false
+ * @prop dense = false
+ * @prop disabled = false
+ * @prop type = 'text'
+ * @prop helptext = ''
+ * @prop helptextPersistent = false
+ * @prop helptextValidationMsg = false
+ */
+export default class Textfield {
+  constructor() {
+    this.id = Textfield.uid();
+  }
+
+  static uid() {
+    if (!this.uidCounter) {
+      this.uidCounter = 0;
+    }
+    return ++this.uidCounter;
+  }
+
+  render(allprops) {
+    const {
+      id,
+      className,
+      helptextPersistent,
+      helptextValidationMsg,
```

Comment:
As you prefer, I chose ```helptextValidationMsg``` because of the name of the class, which is ```mdc-textfield-helptext-validation-msg```, so it was just to stay as close to original as possible, but I can change it if you want.